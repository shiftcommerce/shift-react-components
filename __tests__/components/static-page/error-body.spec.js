// Libraries
import React from 'react'

// Components
import StaticPageErrorBody from '../../../src/components/static-page/error-body'

test('renders correct child elements and classnames', () => {
  const label = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
  const className = 'dummy-classname'

  // act
  const wrapper = mount(
    <StaticPageErrorBody className={className}>
      { label }
    </StaticPageErrorBody>
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Oh no, something went wrong.')
  expect(wrapper).toIncludeText(label)
  expect(wrapper.find('div')).toHaveClassName(className)
})

test('renders correct title and subtitle when passed through as props', () => {
  const strings = {
    title: 'Dummy Title',
    subtitle: 'Dummy Subtitle'
  }

  // act
  const wrapper = mount(
    <StaticPageErrorBody title={strings.title} subtitle={strings.subtitle} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(strings.title)
  expect(wrapper).toIncludeText(strings.subtitle)
})
