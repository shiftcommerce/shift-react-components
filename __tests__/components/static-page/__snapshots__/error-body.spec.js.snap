// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct child elements and classnames 1`] = `
<StaticPageErrorBody
  className="dummy-classname"
>
  <div
    className="dummy-classname c-static-page-error-body"
  >
    <h1>
      Oh no, something went wrong.
    </h1>
    <h2>
      We weren't able to process your request.
    </h2>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit
  </div>
</StaticPageErrorBody>
`;

exports[`renders correct title and subtitle when passed through as props 1`] = `
<StaticPageErrorBody
  subtitle="Dummy Subtitle"
  title="Dummy Title"
>
  <div
    className="c-static-page-error-body"
  >
    <h1>
      Dummy Title
    </h1>
    <h2>
      Dummy Subtitle
    </h2>
  </div>
</StaticPageErrorBody>
`;
